pkg/pillar/zedcloud/deferred.go

```diff
@@ -158,69 +228,100 @@ func (ctx *DeferredContext) removeDeferred(log *base.LogObject, key string) {
 // other Urls, where in other caes, the key is very specific for the object
 //  and object type
 func SetDeferred(zedcloudCtx *ZedCloudContext, key string, buf *bytes.Buffer,
-	size int64, url string, bailOnHTTPErr bool) {
+	size int64, url string, bailOnHTTPErr bool, itemType interface{}) {
 
-	zedcloudCtx.deferredCtx.setDeferred(zedcloudCtx, key, buf, size, url, bailOnHTTPErr)
+	zedcloudCtx.deferredCtx.setDeferred(zedcloudCtx, key, buf, size, url, bailOnHTTPErr, itemType)
 }
 
 func (ctx *DeferredContext) setDeferred(zedcloudCtx *ZedCloudContext,
-	key string, buf *bytes.Buffer, size int64, url string, bailOnHTTPErr bool) {
+	key string, buf *bytes.Buffer, size int64, url string, bailOnHTTPErr bool, itemType interface{}) {
+	ctx.lock.Lock()
+	defer ctx.lock.Unlock()
 
 	log := zedcloudCtx.log
-	log.Functionf("SetDeferred(%s) size %d map %d\n",
+	log.Functionf("SetDeferred(%s) size %d items %d\n",
 		key, size, len(ctx.deferredItems))
 	if len(ctx.deferredItems) == 0 {
 		startTimer(log, ctx)
 	}
-	_, ok := ctx.deferredItems[key]
-	if ok {
-		log.Tracef("Replacing key %s\n", key)
-	} else {
-		log.Tracef("Adding key %s\n", key)
-	}
 	item := deferredItem{
+		added:         time.Now(),
```

Comment:
I think this is wrong.
If object X had a setDeferred 1 hour ago (which didn't get sent yet) and a new setDeferred now, then it shouldn't loose its place in line but be sent based on the 1 hour ago. Otherwise an object with ongoing updates can be starved for an unbounded time.